Add a flag, "disable-ghost-block-fix"

zeanrivera opened the pull request:
Add a flag, "disable-ghost-block-fix", to allow for turning off 
the GhostBlockFix https://github.com/PaperMC/Paper/pull/2396

The reason for this is that this fix re-introduces
this bug https://bugs.mojang.com/browse/MC-156013

This gives server owners a way to opt out of the GhostBlockFix.
It defaults to false, which is keeping the GhostBlockFix.

A friend of mine is able to reproduce this:
- disableGhostBlockFix false: https://www.youtube.com/watch?v=QIuc0EBtMfc
- disableGhostBlockFix true: https://www.youtube.com/watch?v=lWWHeHBkNLA

Spottedleaf commented on Spigot-Server-Patches/0416-Add-a-flag-disableGhostBlockFix-to-allow-for-turning.patch:
```diff
@@ -0,0 +1,50 @@
+                     this.world.a(this.player.getId(), blockposition, j);
+-                    this.player.playerConnection.sendPacket(new PacketPlayOutBlockChange(this.world, blockposition)); // Paper - fixes MC-156852
++                    
++                    // Paper - The GhostBlockFix which fixes MC-156852 reintroduces the nudging bug in MC-156013
```
enclose with //paper start and //paper end

Spottedleaf commented on Spigot-Server-Patches/0416-Add-a-flag-disableGhostBlockFix-to-allow-for-turning.patch:
```diff
@@ -0,0 +1,50 @@
+From c35bd6b639696eeb12865770b86283a96efb7b55 Mon Sep 17 00:00:00 2001
```
this should edit the original patch

zeanrivera committed:
Add a flag, "disable-ghost-block-fix", to allow for turning off the GhostBlockFix https://github.com/PaperMC/Paper/pull/2396

The reason for this is that this fix re-introduces
this bug https://bugs.mojang.com/browse/MC-156013

This gives server owners a way to opt out of the GhostBlockFix.
It defaults to false, which is keeping the GhostBlockFix.

From: Zean Rivera <zean@zean.org>
Date: Sat, 7 Sep 2019 03:11:05 -0700
Subject: [PATCH] Add a flag, "disable-ghost-block-fix", to allow for turning off the GhostBlockFix https://github.com/PaperMC/Paper/pull/2396

The reason for this is that this fix re-introduces
this bug https://bugs.mojang.com/browse/MC-156013

This gives server owners a way to opt out of the GhostBlockFix.
It defaults to false, which is keeping the GhostBlockFix.

BillyGalbreath commented on Spigot-Server-Patches/0406-Fixed-MC-156852.patch:
```diff
@@ -11,18 +11,51 @@ issue in 1.8-1.12.
++                    if (!disableGhostBlockFix) { 
++                        this.player.playerConnection.sendPacket(new PacketPlayOutBlockChange(this.world, blockposition)); // Paper - fixes MC-156852
++                    }
++                    // Paper end
```
Could shorten all this noise down into a single 1 line addition

```
+                    if (!this.world.paperConfig.disableGhostBlockFix) // Paper - The GhostBlockFix which fixes MC-156852 reintroduces the nudging bug in MC-156013
                     this.player.playerConnection.sendPacket(new PacketPlayOutBlockChange(this.world, blockposition)); // Paper - fixes MC-156852
```

zeanrivera replied:
That's fair I suppose. I looked around the rest of the file when making my change and I saw the convention of declaring a boolean and checking it. At a cursory glance, I didn't see other one-line if statements in the file. 

I can change it though if that would adhere to proper convention.

zachbr replied:
Please make this change

zachbr commented on Spigot-Server-Patches/0406-Fixed-MC-156852.patch:
```diff
@@ -11,18 +11,51 @@ issue in 1.8-1.12.
+
+This gives server owners a way to opt out of the GhostBlockFix.
+It defaults to false, which is keeping the GhostBlockFix.
+
```
We do not add duplicate patch headers to files. Remove this. Git does a fine enough job keeping track of who did what without adding multiple headers to files.
